dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api.test/cases/WorkFlowShellAPITest.java

```diff
@@ -0,0 +1,211 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.dolphinscheduler.api.test.cases;
+
+import org.apache.dolphinscheduler.api.test.Constants;
+import org.apache.dolphinscheduler.api.test.core.DolphinScheduler;
+import org.apache.dolphinscheduler.api.test.entity.HttpResponse;
+import org.apache.dolphinscheduler.api.test.entity.LoginResponseData;
+import org.apache.dolphinscheduler.api.test.pages.LoginPage;
+import org.apache.dolphinscheduler.api.test.pages.project.ProjectPage;
+import org.apache.dolphinscheduler.api.test.pages.project.WorkFlowDefinitionPage;
+import org.apache.dolphinscheduler.api.test.pages.project.WorkFlowInstancesPage;
+import org.apache.dolphinscheduler.api.test.pages.security.TenantPage;
+import org.apache.dolphinscheduler.api.test.utils.JSONUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import lombok.extern.slf4j.Slf4j;
+
+@DolphinScheduler(composeFiles = "docker/basic/docker-compose.yaml")
+@Slf4j
+public class WorkFlowShellAPITest {
+    private static final Logger logger = LoggerFactory.getLogger(ProjectAPITest.class);
+
+    private static final String projectName = "case02_wen";
+
+    private static final String projectDesc = "123";
+
+    private static final String workFlowName = "shell123";
+
+    private static final String tenantName = System.getProperty("user.name");
+
+    private static final String user = "admin";
+
+    private static final String password = "dolphinscheduler123";
+
+    private static String sessionId = null;
+
+    ArrayList<Object> localParams = new ArrayList<>();
+
+    ArrayList<Object> resourceList = new ArrayList<>();
+
+    private static String rawScript = "echo 123";
+
+    private static String delayTime = "0";
+
+    private static String description = "";
+
+    private static String environmentCode = "-1";
+
+    private static String failRetryInterval = "1";
+
+    private static String failRetryTimes = "0";
+
+    private static String flag = "YES";
+
+    private static String taskDefinitionRequestDataName = "shell123";
+
+    private static String taskPriority = "MEDIUM";
+
+    private static String taskType = "SHELL";
+
+    private int timeout = 0;
+
+    private static String timeoutFlag = "CLOSE";
+
+    private static String timeoutNotifyStrategy = "";
+
+    private static String workerGroup = "default";
+
+    private static String taskRelationRequestDataName = "";
+
+    private int preTaskCode = 0;
+
+    private int preTaskVersion = 0;
+
+    private static String conditionType = "NONE";
+    private static String executionType = "PARALLEL";
```

Comment:
I think we should create corresponding enumeration types for enumeration values instead of using strings. But I'm not quite sure whether we should do this. PTAL @kezhenxu94

Reply:
> I think we should create corresponding enumeration types for enumeration values instead of using strings. But I'm not quite sure whether we should do this. PTAL @kezhenxu94

Agree.